src/Windows/Avalonia.Win32/WindowImpl.AppWndProc.cs

```diff
@@ -394,6 +395,26 @@ protected virtual unsafe IntPtr AppWndProc(IntPtr hWnd, uint msg, IntPtr wParam,
                         break;
                     }
 
+                case WindowsMessage.WM_CAPTURECHANGED:
+                    {
+                        if (IsMouseInPointerEnabled)
+                        {
+                            break;
+                        }
+                        if (!IsOurWindow(lParam))
```

Comment:
There's a different instance of `IMouseDevice` (and associated `IPointer`) per window.
Consequently, we should probably not check other windows, or pointers from them will keep the capture even though they don't have it anymore.

(Note that having a mouse device per window causes other problems but this isn't the right PR to change that.)

Reply:
If you receive a capture cancellation but you know the new capture is in your application, you can rely on the normal app events to deal with it, and you wouldn't want to terminate it externally. Unless there is also a separate capture element per window?